### PR TITLE
Add support for configuring pods as arbiterOnly based on labels

### PR DESCRIPTION
**Pull request by luqasn**
Still needs documentation for the new env variable MONGO_SIDECAR_POD_LABELS_ARBITER. This is something we needed in our cluster, maybe it is also of interest to others. What do you think about the approach of additional labels used for selecting a subset of the pods?

### CODE REVIEW

File: src/lib/worker.js [hunk 1]
```diff
@@ -213,7 +213,9 @@ var invalidReplicaSet = function(db, pods, done) {
 var podElection = function(pods) {
   //Because all the pods are going to be running this code independently, we need a way to consistently find the same
   //node to kick things off, the easiest way to do that is convert their ips into longs and find the highest
-  pods.sort(function(a,b) {
+  pods
+  .filter(function(pod) { return !k8s.podContainsLabels(pod, config.mongoPodLabelCollectionArbiter); })
```
Comment (luqasn) [inline src/lib/worker.js]: I think this function is currently not working as intended. The caller expects the pods array it passes into the function to be sorted in-place, which the filter prohibits by creating a copy. I need to fix that.

Comment (cvallance) [inline src/lib/worker.js]: Yep, you got.

```
pods = pods.filter(function(pod) { return !k8s.podContainsLabels(pod, config.mongoPodLabelCollectionArbiter); });
//Now you can sort and carry on as you were before.
```

**Comment by cvallance** [comment]
Nice, looks good.

> What do you think about the approach of additional labels used for selecting a subset of the pods?

Initially I thought that it would be confusing to consumers in that they would have to understand arbiters would have to satisfy `MONGO_SIDECAR_POD_LABELS` **and** `MONGO_SIDECAR_POD_LABELS_ARBITER` ... not that hard to understand, but people don't like reading documentation and are use to specific behaviours when seeing different config patterns. If someone saw them and thought that they were completely independent, could you blame them? Is that how you'd expect it to work given past experiences with similar libraries?

But in saying that... if you wanted to make them completely separate, you'd need to change a lot more code. And, really, if you are going to start setting arbiters, you should be reading the documentation to see exactly how it works.

So in short, I'm cool with either approach as long as they are documented to make it clear 😄  ... maybe a row in the configuration table and a separate little blurb on setting up arbiters.

**Comment by luqasn** [comment]
Closing this in favour of #73, thanks @ericbaum, kind of lazied out of this after I stopped using the lib :)
